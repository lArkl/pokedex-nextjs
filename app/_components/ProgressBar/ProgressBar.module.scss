.container {
  background-color: var(--color-white-2);
  border-radius: 0.55rem;
  width: 95%;
  padding: 0.1rem 0.2rem;
  display: flex;
  align-items: center;
}

.bar {
  background-color: var(--color-purple-1);
  border-radius: 0.4rem;
  height: 80%;
  padding: 0.1rem;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  transform-origin: left;
  animation: grow 1.5s;
}

@keyframes grow {
  0% {
    transform: scaleX(0.1);
  }
}

.value {
  background-color: white;
  border-radius: 50%;
  height: 0.65rem;
  width: 0.65rem;
  margin-right: 0.02rem;
}
