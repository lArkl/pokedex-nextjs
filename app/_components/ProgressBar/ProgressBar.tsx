import { FC } from 'react'
import styles from './ProgressBar.module.scss'

export interface ProgressBarProps {
  value: number
  max?: number
}

const ProgressBar: FC<ProgressBarProps> = ({ value, max = 100 }) => {
  const width = `${Math.floor((value / max) * 100)}%`
  return (
    <div className={styles.container} role="progressbar">
      <div className={styles.bar} style={{ width }}>
        <div className={styles.value}></div>
      </div>
    </div>
  )
}

export default ProgressBar
