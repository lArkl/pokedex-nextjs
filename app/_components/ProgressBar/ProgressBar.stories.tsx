import type { Meta, StoryObj } from '@storybook/react'

import ProgressBar from './ProgressBar'

const meta = {
  title: 'Atoms/ProgressBar',
  component: ProgressBar,
  parameters: {
    layout: 'centered',
  },
  decorators: [(story) => <div style={{ width: '10rem' }}>{story()}</div>],
  tags: ['autodocs'],
  argTypes: {},
} satisfies Meta<typeof ProgressBar>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
  args: {
    value: 150,
    max: 255,
  },
}
