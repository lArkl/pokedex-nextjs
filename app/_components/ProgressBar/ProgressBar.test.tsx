import { render } from '@testing-library/react'
import ProgressBar from './ProgressBar'

describe('ProgressBar', () => {
  it('renders component', async () => {
    const { asFragment } = render(<ProgressBar value={0} />)

    expect(asFragment()).toMatchSnapshot()
  })
})
