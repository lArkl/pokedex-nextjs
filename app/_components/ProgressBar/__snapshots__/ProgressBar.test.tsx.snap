// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ProgressBar > renders component 1`] = `
<DocumentFragment>
  <div
    class="_container_98c422"
    role="progressbar"
  >
    <div
      class="_bar_98c422"
      style="width: 0%;"
    >
      <div
        class="_value_98c422"
      />
    </div>
  </div>
</DocumentFragment>
`;
